### ISSUE TITLE
Cannot build production Android build using .OBJ/.MTL files

### ISSUE THREAD

**Issue by jlyonsmith** [viromedia/viro#217]
I'm hitting a similar issue to #169.   I'm unable to create a production Android build that includes `.obj` and `.mtl` files.  I get the following build error message from the `gradlew` script:

```
:app:processReleaseManifest
:app:processReleaseResources
res/drawable-mdpi-v4/js_res_clipboard_clipboard.obj:0: error: Resource entry js_res_clipboard_clipboard is already defined.
res/drawable-mdpi-v4/js_res_clipboard_clipboard.mtl:0: Originally defined here.

res/drawable-mdpi-v4/js_res_hardhat_hardhat.obj:0: error: Resource entry js_res_hardhat_hardhat is already defined.
res/drawable-mdpi-v4/js_res_hardhat_hardhat.mtl:0: Originally defined here.

res/drawable-mdpi-v4/js_res_question_question.obj:0: error: Resource entry js_res_question_question is already defined.
res/drawable-mdpi-v4/js_res_question_question.mtl:0: Originally defined here.

:app:processReleaseResources FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processReleaseResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt
```

This also happens in my [TestViroAR](https://github.com/jlyonsmith/TestViroAR) repository so you can easily reproduce the issue using that project.  I've tried moving the `.mtl` to a different directories/filenames and removing the `mtllib` directive from inside the `.obj` files, but that doesn't work.  Google doesn't provide much assistance on the `aapt` tool.

I'm going to try converting the `.obj` to `.fbx` and converting to `.vbx`, but ideally I'd like for the project to be buildable on a Windows if possible.

Any ideas?

**Comment by manbod** [comment]
Hey @jlyonsmith,
The error you are getting is a limitation / by-design of how android stores application's resources during compile time - You cannot store resource files with same names but different extensions in android (https://stackoverflow.com/questions/24404722/how-to-store-resource-file-with-same-name-but-different-extensions-in-android). That's why you are seeing the `error: Resource entry js_res_clipboard_clipboard is already defined.` Can you try renaming just your OBJ files by adding a `_obj` to the file name itself? So that after compilation the OBJ and the MTL end up having different names in resources. (Keep the MTL file names as they are - since changing them might require changing the `mtllib` directive inside the `.obj` file as well).

Let us know if that works.

P.S. - I noticed in your `TestViroAR` repository you are using  `"react-viro": "^2.4.0"` with `"react-native": "^0.51.1"`. We have tested our most recent release `v2.5.1` with react-native `v0.49.3` ([ViroReact Releases](https://docs.viromedia.com/docs/releases)). While we are looking at updating our compatibility with a newer version of react-native, in an upcoming release, your mileage may vary with newer versions of react-native :-) . Good luck!

**Comment by jlyonsmith** [comment]
Thanks @manbod.  That was problem. I thought I had tried that, but it turns out I needed to delete the `android/app/build/` directory between attempts for `gradlew` to pick up the file name changes.

Thanks for the speedy response!

p.s. I'll keep an eye out for weird behavior caused by the version mismatch.  So far the only issue that I discovered was on the Android side when I tried to upgrade build tools to the latest versions.  The AR view would come up as a black screen.  I'll file a bug and a repro if I hit the problem again.
